web-playground/e2e/sample-tester.test.ts

```diff
@@ -0,0 +1,82 @@
+import { FrameWaitForFunctionOptions, Page } from "puppeteer";
+import * as fs from 'fs';
+import * as path from 'path';
+
+describe('test all grammars with puppeteer', () => {
+    beforeEach(async () => {
+        await page.goto('http://localhost:3000/playground');
+    });
+
+    it('can format', async () => {
+        const rootDir = path.join(__dirname, "../../");
+        const inputDir = path.join(rootDir, "topiary/tests/samples/input/");
+        const expectedDir = path.join(rootDir, "topiary/tests/samples/expected/");
+        const queryDir = path.join(rootDir, "languages/");
+
+        for (let inputFileName of await fs.promises.readdir(inputDir)) {
+            const inputPath = path.join(inputDir, inputFileName);
+            const expectedPath = path.join(expectedDir, inputFileName);
+            const queryFileName = inputFileName.replace(/\..*$/, ".scm");
+            const queryPath = path.join(queryDir, queryFileName);
+
+            console.log(`Testing ${inputPath} ${expectedPath} ${queryPath}`);
+
+            const encoding = "utf8";
+            const input = await fs.promises.readFile(inputPath, encoding);
+            const expected = await fs.promises.readFile(expectedPath, encoding);
+            const query = await fs.promises.readFile(queryPath, encoding);
+
+            await testInputFile(input, expected, query);
+        }
+    }, 20000);
```

Comment:
Is this `20000` a timeout, presumably in milliseconds? It might be worth adding it as a top-level constant, so it looks less like a magic number.

Reply:
Fixed.